### PR TITLE
CLOUDP-97077: [Milestone - Atlas] Update performanceAdvisor with new options

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-97077

Added E2E test: [atlas_performance_advisor_e2e](https://evergreen.mongodb.com/task/mongocli_master_e2e_atlas_clusters_atlas_performance_advisor_e2e_patch_e0481661d5a15c20ffc919b2bc90ea686bda8d28_613b6af7d6d80a282336ff05_21_09_10_14_26_01)

Added E2E test: [atlas_gov_performance_advisor_e2e](https://evergreen.mongodb.com/task/mongocli_master_e2e_atlas_gov_clusters_atlas_gov_performance_advisor_e2e_patch_e0481661d5a15c20ffc919b2bc90ea686bda8d28_613b6af7d6d80a282336ff05_21_09_10_14_26_01)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas performanceAdvisor sot disable -P test
Atlas management of the slow operation disabled
```

```
./bin/mongocli atlas performanceAdvisor sot enable -P test
Atlas management of the slow operation enabled
```

### CODE REVIEW

File: internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go
Comment (gssbzn) [inline internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go:26]: I'll let docs chime in but "manage the management" is a bit too much if you ask me

File: internal/cli/performanceadvisor/slowoperationthreshold/disable.go
Comment (gssbzn) [inline internal/cli/performanceadvisor/slowoperationthreshold/disable.go:34]: [q] do you need this? doesn't seem used

Comment (andreaangiolillo) [reply internal/cli/performanceadvisor/slowoperationthreshold/disable.go:34]: no, `cli.OutputOpts` is a left-over

File: internal/cli/performanceadvisor/slowoperationthreshold/enable.go
Comment (gssbzn) [inline internal/cli/performanceadvisor/slowoperationthreshold/enable.go:33]: same question, is this needed?

Comment (andreaangiolillo) [reply internal/cli/performanceadvisor/slowoperationthreshold/enable.go:33]: no

Comment (andreaangiolillo) [reply internal/cli/performanceadvisor/slowoperationthreshold/enable.go]: updated

File: internal/cli/performanceadvisor/slowoperationthreshold/disable.go
Comment (andreaangiolillo) [reply internal/cli/performanceadvisor/slowoperationthreshold/disable.go]: updated

**Review by gssbzn — APPROVED** [review]
LGTM  if LGTDOCS thanks for adding all the testing 💯

File: internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go
Comment (melissamahoney-mongodb) [reply internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go:26]: Since there's only two commands, we can probably just be explicit?
```suggestion
		Short:   "Enable or disable Atlas management of the slow operation threshold for your cluster.",
```

File: internal/cli/performanceadvisor/slowquerylogs/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowquerylogs/list.go:78]: I think this description is wrong, this is for logs not namespaces right?
```suggestion
		Short:   "Return log lines for slow queries that the Performance Advisor and Query Profiler identified.",
		Long: `The Performance Advisor monitors queries that MongoDB considers slow and suggests new indexes to improve query performance. The threshold for slow queries varies based on the average time of operations on your cluster to provide recommendations pertinent to your workload.
You must have one of the following roles to run this command:
Project Owner access
Project Data Access Admin access
Project Data Access Read/Write access
Project Data Access Read Only access`,
		Example: "mongocli atlas performanceAdvisor slowQueryLogs ls --processName atlas-111ggi-shard-00-00.111xx.mongodb.net:27017",
```

File: internal/cli/performanceadvisor/suggestedindexes/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/suggestedindexes/list.go:79]:
```suggestion
		Short:   "Retrieves suggested indexes for collections experiencing slow queries.",
		Long:    "The Performance Advisor monitors queries that MongoDB considers slow and suggests new indexes to improve query performance.",
```

File: internal/cli/performanceadvisor/namespaces/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/namespaces/list.go:71]: Is the "up to 20 namespaces" part still accurate? This comes from our API docs:
```suggestion
		Short:   "Retrieves up to 20 namespaces for collections experiencing slow queries for a specified host.",
		Long:    "Namespaces appear in the following format: {database}.{collection}.",
		Example: "$ mongocli atlas performanceAdvisor namespaces list --processName atlas-111ggi-shard-00-00.111xx.mongodb.net:27017",
```

File: internal/cli/performanceadvisor/namespaces/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/namespaces/list.go:73]:
```suggestion
```

File: internal/cli/performanceadvisor/slowoperationthreshold/disable.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowoperationthreshold/disable.go:56]:
```suggestion
		Short:   "Disable the application-managed slow operation threshold for your project.",
```

File: internal/cli/performanceadvisor/slowoperationthreshold/disable.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowoperationthreshold/disable.go:58]:
```suggestion
```

File: internal/cli/performanceadvisor/slowoperationthreshold/enable.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowoperationthreshold/enable.go:58]:
```suggestion
		Short:   "Enable the application-managed slow operation threshold for your project.",
		Long:    "The slow operation threshold determines which operations are flagged by the Performance Advisor and Query Profiler. When enabled, the application uses the average execution time for operations on your cluster to determine slow-running queries. As a result, the threshold is more pertinent to your cluster workload. Application-managed slow operation threshold is enabled by default for dedicated clusters (M10+).",
```

File: internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowoperationthreshold/slow_operation_threshold.go:26]:
```suggestion
		Short:   "Enable or disable management of the slow operation threshold for your cluster.",
```

File: internal/cli/performanceadvisor/slowquerylogs/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/slowquerylogs/list.go:85]:
```suggestion
		Long: `The Performance Advisor monitors queries that MongoDB considers slow and suggests new indexes to improve query performance. The threshold for slow queries varies based on the average time of operations on your cluster to provide recommendations pertinent to your workload.

You must have one of the following roles to run this command:
Project Owner access
Project Data Access Admin access
Project Data Access Read/Write access
Project Data Access Read Only access`,
```

File: internal/cli/performanceadvisor/suggestedindexes/list.go
Comment (melissamahoney-mongodb) [inline internal/cli/performanceadvisor/suggestedindexes/list.go:81]:
```suggestion
```

**Review by melissamahoney-mongodb — CHANGES_REQUESTED** [review]
@andreaangiolillo I didn't realize this text would be reused for CM and OM commands too, so I took another pass to make them more generic. I removed the examples until we figure out a better way to do them (because Atlas examples were showing on CM and OM pages)

**Review by melissamahoney-mongodb — APPROVED** [review]

**Review by colm-quinn — APPROVED** [review]
LGTM

**Pull request merged**
